README.md

```diff
@@ -39,59 +51,85 @@ The function `issuesToLineNumbers` invokes both of them: it will build the tree
+- `['spec', 'template', 'spec', 'containers[0]', 'nonExistingResource', 'securityContext', 'capabilities']`
 
 It will return the line number of the first element in the containers array (because nonExistingResource does not exist).
 
-For the received path: 
+For the received path:
 
-- ```['spec', 'template', 'spec', 'containers[0]', 'resources', 'securityContext', 'capabilities']```
+- `['spec', 'template', 'spec', 'containers[0]', 'resources', 'securityContext', 'capabilities']`
 
 It will return the line number of 'capabilities'.
 
+#### Elements with array:
 
-### Elements with array:
 ---
 
-Until now, the paths received in the Cloud Config issues were `containers[snyky1]`, where `snyky1` was the value of the `name` property in one of the objects in `containers`.  
+Until now, the paths received in the Cloud Config issues were `containers[snyky1]`, where `snyky1` was the value of the `name` property in one of the objects in `containers`.
 
 We are supporting both `containers[snyky1]` and `containers[0]`, while the new issues will be in the format of `containers[0]`.
 
-The piece of code that creates the paths is creating elements like `containers[0]`, but in cases of `containers[snyky1]`, it goes over the elements of the array and looks for a sub-element with key: `name` and value `snyky1`.  
+The piece of code that creates the paths is creating elements like `containers[0]`, but in cases of `containers[snyky1]`, it goes over the elements of the array and looks for a sub-element with key: `name` and value `snyky1`.
 
+#### **Paths starting with 'input'**
 
-### **Paths starting with 'input'**  
 ---
 
-For example:  
+For example:
+
+`['input', 'spec', 'template', 'spec', 'containers[0]', 'resources', 'securityContext', 'capabilities']`
 
-```['input', 'spec', 'template', 'spec', 'containers[0]', 'resources', 'securityContext', 'capabilities']```
+The input value will be removed and the path we are looking for will be like this:
 
-The input value will be removed and the path we are looking for will be like this:  
+`['spec', 'template', 'spec', 'containers[0]', 'resources', 'securityContext', 'capabilities']`
 
-```['spec', 'template', 'spec', 'containers[0]', 'resources', 'securityContext', 'capabilities']```
+#### **Yaml DocId:**
 
-### **Yaml DocId:**
 ---
 
-- In the case that the files are JSON or single Yaml - the `DocId` will be `0`.  
+- In the case that the files are JSON or single Yaml - the `DocId` will be `0`.
 - In the case of a multi-document file - the `DocId` will be according to the order of the documets.
 
+#### Line numbers
 
-### Line numbers
 ---
+
 Are 1 based!
 
+#### Keys - not value
 
-### Keys - not value
 ---
 
 We are looking for the key in the path and not the value.
 
 For example , `drop` may have multiple values as an array of strings. We can show `drop[0]` as the first array of values but not `drop['192.168.0.1']`.
+
+## Parse issue path for smaller components.
```

Comment:
Main thing here is that we store the serialized path, seperated with dots, as a string in our database but our functions expect it to be broken into an array.

```suggestion
## Parse serialized issue path for use in `getLineNumber()`
```